server/fileupdater.mjs

```diff
@@ -349,7 +349,7 @@ function v9NumericStringSort(routine) {
 
 function v10GridOffset(properties) {
   const grid = properties.grid;
-  if (!grid)
+  if (!grid || typeof grid != 'object' || !grid.length)
```

Comment:
`!grid.length` is problematic here.

Reply:
Oh is it? falsey grid.length should be skippable no?

Reply:
Should be `!Object.prototype.hasOwnProperty(grid,'length')`

Reply:
Could just check that grid is an array with `Array.isArray(grid)`

Reply:
for in also works on arrays, and I know we had to use it in a few other places so that the code linter was satisfied that user input couldn't cause unbounded iteration lengths.

Reply:
But also happy to have the array check if you prefer. Mostly I wanted to fix the crash.

Reply:
I don't thing for in hurts. Code using grid iterates with for of.